.github/workflows/release_github.yml

```diff
@@ -4,33 +4,79 @@ on:
   workflow_dispatch:
     inputs:
       version_level:
-        description: "Semantic version level increase."
+        description: Semantic version level increase
         required: true
         type: choice
+        default: patch
         options:
           - patch
           - minor
           - major
-
+          
 permissions:
   contents: write
-  pull-requests: write
   packages: read
+  statuses: read
+  checks: read
+  pull-requests: write
+  actions: read
+  repository-projects: read
+
+env:
+  GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+  GH_PAGER: cat
 
 jobs:
+  checks:
+    name: Check requirements
+    runs-on: "ubuntu-latest"
+    steps:
+      - name: Fail if main branch was selected
+        if: ${{ github.ref_name == 'main' }}
+        run: |
+          echo "Cannot release from main branch, please select valid release branch."
+          exit 1
+
+      - name: Checkout repository
+        if: ${{ github.ref_name != 'develop' }}
+        uses: actions/checkout@v4
+
+      - name: Check if PR exists
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }}
+
+      - name: Check if PR base is main
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }} --json baseRefName -q 'if .baseRefName == "main" then "PR base is main" else error("PR base is not main") end'
+        
+      - name: Check if PR is mergeable
+        if: ${{ github.ref_name != 'develop' }}
+        run: gh pr view ${{ github.ref_name }} --json mergeable -q 'if .mergeable == "MERGEABLE" then "PR is mergeable" else error("PR is not mergeable") end'
+
+      - name: Check whether PR checks pass(ed)
+        # note that this assumed there are checks to be passed; this fails if there are no checks, which is no
+        # relevant to this repo
+        if: ${{ github.ref_name != 'develop' }}
+        run: |
+          gh pr checks ${{ github.ref_name }} --watch --fail-fast
+          gh pr checks ${{ github.ref_name }} --json state -q 'if . | length == 0 then "No checks found." elif map(.state == "SUCCESS") | all then "All checks passed" else error("Not all checks passed") end'            
+
   test_python_releases:
+    needs: checks
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
-      matrix: 
+      matrix:
         os: ["ubuntu-latest"]
```

Comment:
I don't think there's anything specific to linux in this package, right? Maybe it would be nice to add OSX and windows to the matrix as well, to ensure cross-platform compatibility?

Reply:
Yes! And no. But mostly yes.

Doing all versions on all platforms is (unnecessarily?) expensive and slow. Should we test the lowest supported version on all platforms, and up to the newest supported version on linux?

Reply:
I think given that it's only done rarely (upon release) it's fine to have a large matrix.
If you feel it's overkill, I would leave it like it was. I think it would create more confusion than clarity to create some versions on some platforms and all versions on another (plus would again become quite verbose).